### DISCUSSION TITLE
Why is withdraw function payable?

### DISCUSSION THREAD

**Discussion by jasjyotsinghjaswal** [smartcontractkit/full-blockchain-solidity-course-py#134] [Ideas]
Do we really need to make withdraw function payable? Because based on what you explained earlier payable is only required when sending ETH to contract but here we are transferring eth from contract balance to senders account?

**Comment by PatrickAlphaC** [comment]
Great question! Let me answer with another question... does your code compile when you remove the `payable` keyword?

**Reply by jasjyotsinghjaswal** [reply]
Well indeed yes, when i remove the payable keyword the function still compiles. Thanks Anyways and I am perhaps asking too many questions on this forum I know but its just to get my understanding correct . Regardless its such a great course thank you so much for making this content and that too for free

**Reply by PatrickAlphaC** [reply]
Asking questions is exactly what you should be doing. Thank you!

**Comment by PhyoLinMg** [comment]
From what i have learned, smart contracts are like banks,(not exactly) but let's assume it. When you withdraw,transfer ETH, it needs to be payable like any other banks you would find.
**If I had said something wrong, don't hesitate to correct me.**
